src/tribler-gui/tribler_gui/widgets/tablecontentmodel.py

```diff
@@ -44,7 +44,9 @@ def __init__(self, parent=None):
         self.sort_desc = True
         self.saved_header_state = None
         self.saved_scroll_state = None
+        self.qt_object_destroyed = False
 
+        connect(self.destroyed, self.on_destroy)
```

Comment:
:+1: 

Excellent solution! I did not know about this signal.
I guess that's the closest thing we can have to a proper C++ destructor...